src/DebounceInput.js

```diff
@@ -107,17 +83,23 @@ const DebounceInput = React.createClass({
     } else {
       onChange('');
     }
+
+    if (this.notify.cancel) {
```

Comment:
@MatthewMaclean found the problem with duplication of onChange. Debounced callback still worked. Your History example revealed this.